### ISSUE TITLE
Support reading the sunpy configuration file from global directories

### ISSUE THREAD

**Issue by Cadair** [sunpy/sunpy#6407]
### Describe the feature

Currently we support a per-user config file.

Sometimes on a multi-user system, i.e. a cluster, it could be useful to have a system default. Which would, for example, allow you to set the download directory to a faster, or shared file system by default.

### Proposed solution

Appdirs has a `site_config_dir` which returns the platform specific global config directory, if we modify 

https://github.com/sunpy/sunpy/blob/33f46bfcf320137e91e4cd3262e5fb6844ed8474/sunpy/util/config.py#L62-L80

to look for a config file in the site directory if no user specific config file is found this would work.

**Comment by tlml** [comment]
I've started working on this, almost done. Couple of questions/comments:

- There was discussion in Element that the SUNPY_DOWNLOADDIR shouldn't be used, should it be deprecated?

- Is there a need to warn the user about database and remote_data_manager database conflicts by different users (in the documentation)? I had a quick look at the code and it looks they are already protected but can't be sure... (I guess a conflict could take place also for one user...)

-  I didn't change the copy_default_config to take the site configuration into account, as it looks more trouble than worth... I did put in a warning if there is a site config that the user has to take it into account manually, and will add to documentation.
